_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,461 @@
+            }
+        ]
+    }
+}
+```
+
+
+## Examples 
+
+The following example connector requests show how to create a connector with supported third-party tools.
+
+
+### OpenAI chat connector
+
+The following example creates a standalone OpenAI chat connector. The same options can be used for an internal connector under the `connector` parameter:
+
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "OpenAI Chat Connector",
+    "description": "The connector to public OpenAI model service for GPT 3.5",
+    "version": 1,
+    "protocol": "http",
+    "parameters": {
+        "endpoint": "api.openai.com",
+        "model": "gpt-3.5-turbo"
+    },
+    "credential": {
+        "openAI_key": "..."
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "url": "https://${parameters.endpoint}/v1/chat/completions",
+            "headers": {
+                "Authorization": "Bearer ${credential.openAI_key}"
+            },
+            "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": ${parameters.messages} }"
+        }
+    ]
+}
+```
+
+After creating the connector, you can retrieve the `task_id`, deploy the model, and use the Predict API, similar to a standalone connector.
+
+
+### AWS SageMaker
+
+The following example creates a standalone Amazon SageMaker connector. The same options can be used for an internal connector under the `connector` parameter:
+
+```json
+POST /_plugins/_ml/connectors/_create
+{
+    "name": "sagemaker: embedding",
+    "description": "Test connector for Sagemaker embedding model",
+    "version": 1,
+    "protocol": "aws_sigv4",
+    "credential": {
+        "access_key": "...",
+        "secret_key": "...",
+        "session_token": "..."
+    },
+    "parameters": {
+        "region": "us-west-2",
+        "service_name": "sagemaker"
+    },
+    "actions": [
+        {
+            "action_type": "predict",
+            "method": "POST",
+            "headers": {
+                "content-type": "application/json"
+            },
+            "url": "https://runtime.sagemaker.${parameters.region}.amazonaws.com/endpoints/lmi-model-2023-06-24-01-35-32-275/invocations",
+            "request_body": "[\"${parameters.inputs}\"]"
+        }
+    ]
+}
+```
+
+The `credential` parameter contains the following options reserved for `aws-sigv4` authentication:
+
+- `access_key`: Required. Provides the access key for the AWS instance.
+- `secret_key`: Required. Provides the secret key for the AWS instance.
+- `session_token`: Optional. Provides a temporary set of credentials for the AWS instance.
+
+The `paramaters` section requires the following options when using `aws-sigv4` authentication:
+
+- `region`: The AWS Region in which the AWS instance is located.
+- `service_name`: The name of the AWS service for the connector.
```

Comment:
As all items defined in `parameters` will be overridable and visible (user can see parameters with get connector API). If user don't want this, they can move these two reserved parameters for `aws_sigv4` to `credential`.